### PR TITLE
Feature/new db connection popup

### PR DESCRIPTION
**Pull request by delieva**
**Description:** Сreated two popups for DB connection and creating user

**Link to Trello ticket:** https://trello.com/c/DLxjdKHS https://trello.com/c/OWSwieS4

### CODE REVIEW

File: src/components/User/newUserPopup/newUserPopup.module.scss [hunk 1]
```diff
@@ -0,0 +1,52 @@
+@import "./../../../assets/variables";
+
+.loginForm {
```
Comment (MishaHnatyshyn) [inline src/components/User/newUserPopup/newUserPopup.module.scss]: Why loginForm?

File: src/components/User/newUserPopup/newUserPopup.module.scss [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+.error {
+  margin: 0 auto 25px auto;
+  width: 320px;
+  transition: opacity .5s ease-in-out;
+}
+
+.line {
+  width: 65px;
+  height: 5px;
+  background-color: #007eff;
```
Comment (MishaHnatyshyn) [inline src/components/User/newUserPopup/newUserPopup.module.scss]: move to variables ?

File: src/components/User/newUserPopup/NewUserCaption/index.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import styles from './newUserCaption.module.scss';
+import Caption from '../../../shared/Caption';
+
+const newUserCaption = () => (
```
Comment (MishaHnatyshyn) [inline src/components/User/newUserPopup/NewUserCaption/index.js:5]: The component name should start from the upper-case character

File: src/components/shared/Input/textField.module.scss [hunk 4]
```diff
@@ -30,6 +30,15 @@
...
+.noImage {
+  border: none;
+  outline: none;
+  width: 100%;
+  border-bottom: 2px solid #cbcbcb;
+  padding: 10px 0;
+  font: $text-regular;
+}
+
 ::-webkit-input-placeholder { /* Edge */
```
Comment (MishaHnatyshyn) [inline src/components/shared/Input/textField.module.scss:42]: Why do we need this?

File: src/config/endpoints.js [hunk 5]
```diff
@@ -10,3 +10,4 @@ export const ADMIN_MODELS_ENDPOINT = '/models';
 export const ADMIN_USERS_COUNT_ENDPOINT = '/users/count';
 export const ADMIN_MODELS_COUNT_ENDPOINT = '/models/count';
 export const ADMIN_CONNECTIONS_COUNT_ENDPOINT = '/connections/count';
+export const ADMIN_CONNECTIONS_CREATE_ENDPOINT = '/connections/count';
```
Comment (MishaHnatyshyn) [inline src/config/endpoints.js]: Is this correct endpoint?

File: src/store/user/selectors.js [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import { compose, prop } from 'lodash/fp';
 
 const root = (state) => state.user;
-export const getUsername = compose(prop('username'), root);
+compose(prop('username'), root);
```
Comment (MishaHnatyshyn) [inline src/store/user/selectors.js]: What is this ?

**Review by MishaHnatyshyn — CHANGES_REQUESTED** [review]

File: src/components/shared/Input/textField.module.scss (hunk 4 above)
Comment (delieva) [reply src/components/shared/Input/textField.module.scss:42]: We need this to have the ability to customize the input field

File: src/components/UsersTable/UsersTableRow/index.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    styles.error,
+    isError ? styles.visible : styles.hidden
+  ], [isError]);
+  return (
+    <div className={styles.loginForm}>
+    
+    </div>
+  );
+};
+
```
Comment (MishaHnatyshyn) [inline src/components/UsersTable/UsersTableRow/index.js]: what is this ?

File: src/components/User/newUserPopup/index.js [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+  okButtonType,
+  cancelButton,
+  onClose,
+}) => {
+  const formHandler = useMemo(
+    () => preventDefaultHandler(submitForm),
+    [submitForm]
+  );
+  const alertClassesPassLen = useMemo(() => [
+    errorMessage === 'Password must be 6 or mor characters' ? styles.visible : styles.hidden,
```
Comment (MishaAfanasiuk) [inline src/components/User/newUserPopup/index.js:61]: Not the best idea to compare error with string
Move it to some constant, mb, it is already there?

File: src/utils/http.js [hunk 9]
```diff
@@ -29,15 +29,29 @@ const createHeaderWithBearerToken = (token) => ({
...
 });
 
+// export const sendWithUserAuthToken = (httpMethod) => (url, config = {}) => {
+//   const token = getToken(store.getState());
+//   const authHeader = token ? createHeaderWithBearerToken(token) : {};
+//   return httpMethod(`${API_URL}${url}`, merge(config, authHeader))
+//     .then((res) => res.data)
+//     .catch(handleUnAuthorizedError);
+// };
+
```
Comment (MishaHnatyshyn) [inline src/utils/http.js]: Delete this comments

File: src/store/connection/actions.js [hunk 10]
```diff
@@ -76,3 +100,38 @@ export const getConnectionsCount = () => async (dispatch, getState) => {
...
+        username: name,
+        password,
+        typeId: Number(type),
+      }
+    });
+    dispatch(setConnections(data));
+  } catch (e) {
+    dispatch(createConnectionFailure());
+  }
+};
```
Comment (MishaHnatyshyn) [inline src/store/connection/actions.js]: How do you handle errors ?

File: src/store/connection/actions.js [hunk 11]
```diff
@@ -76,3 +100,38 @@ export const getConnectionsCount = () => async (dispatch, getState) => {
...
+      port,
+      nameDB,
+      username: name,
+      password,
+      type,
+      nameConnection,
+    }
+  } = getState();
+
+  if (!name || !password || !host || !port || !nameDB || !type || !nameConnection) {
```
Comment (MishaAfanasiuk) [inline src/store/connection/actions.js:147]: Not the best way to check fields
It would be better to  have an array of fields and use some find or etc method

File: src/utils/http.js (hunk 9 above)
Comment (MishaAfanasiuk) [inline src/utils/http.js]: Do we need this?

File: src/components/newDBConnectionPopup/index.js [hunk 12]
```diff
@@ -0,0 +1,189 @@
...
+      unmountOnExit
+    >
+      <div className={styles.popupWrapper}>
+        <div className={styles.blur} onClick={backgroundHandler} />
+        <div className={`${styles.popup}`}>
+          <div className={styles.header}>
+            <div className={styles.cross} onClick={onClose}>
+              <img src="/images/Popup/cross.png" alt="" />
+            </div>
+          </div>
```
Comment (MishaHnatyshyn) [inline src/components/newDBConnectionPopup/index.js]: You use this code twice in both popups, could you move common code in some base component ?

Comment (MishaHnatyshyn) [reply src/components/newDBConnectionPopup/index.js]: Can't you just use this component for popup layout ? https://github.com/MishaHnatyshyn/Datalyzer/blob/develop/src/components/shared/BasePopup/index.js

File: src/components/shared/Input/textField.module.scss [hunk 13]
```diff
@@ -44,7 +53,7 @@
 
 @media all and (max-width: $small-mobile-width) {
   .input {
-    width: 180px;
+    width: 80%;
```
Comment (VladaZhmurkevych) [inline src/components/shared/Input/textField.module.scss:56]: Won't it affects all the other inputs on different pages? Maybe it is better to pass className "classes" with a width of 80%?

**Review by MishaHnatyshyn — CHANGES_REQUESTED** [review]

File: src/components/shared/Input/index.js [hunk 14]
```diff
@@ -12,7 +13,7 @@ const Input = ({
   return (
     <div className={styles.container}>
       <input
-        className={styles.input}
+        className={classes}
```
Comment (VladaZhmurkevych) [inline src/components/shared/Input/index.js:15]: Why did you delete "styles.input"? I think it is better to write "classnames(styles.input, classes)"

File: src/components/shared/Input/index.js [hunk 15]
```diff
@@ -23,11 +24,16 @@ const Input = ({
   );
 };
 
+Input.defaultProps = {
+  classes: [styles.input]
```
Comment (VladaZhmurkevych) [inline src/components/shared/Input/index.js:27]: we should use this style all the time, not when there is not any other class

File: src/components/shared/BasePopup/base.popup.module.scss [hunk 16]
```diff
@@ -34,8 +34,8 @@
 }
 
 .popup {
-  max-width: 410px;
-  min-width: 320px;
+  max-width: 600px;
+  min-width: 450px;
```
Comment (VladaZhmurkevych) [inline src/components/shared/BasePopup/base.popup.module.scss:39]: These changes can affect others pop up

File: src/components/shared/BasePopup/base.popup.module.scss [hunk 17]
```diff
@@ -70,3 +70,41 @@
...
+select {
+  width: 100%;
+  height: 30px;
+  background-color: white;
+  -webkit-appearance: none;
+  -moz-appearance: none;
+  padding-left: 5px;
+  font: normal 18px "Montserrat";
+  margin: 20px 0;
+}
```
Comment (VladaZhmurkevych) [inline src/components/shared/BasePopup/base.popup.module.scss]: Why did you write styles to your pop up in the file "Base Popup", which is used for different popups?

File: src/store/adminUsers/types.js [hunk 18]
```diff
@@ -10,5 +10,14 @@ export const CREATE_SUCCESS = 'ADMIN_USERS::CREATE_SUCCESS';
...
 export const CHANGE_SEARCH_INPUT = 'ADMIN_USERS::CHANGE_SEARCH_INPUT';
+export const FORM_USER_TYPE_INPUT_VALUE = 'ADMIN_USERS::FORM_USER_TYPE_INPUT_VALUE';
+export const FORM_DESCRIPTION_INPUT_VALUE = 'ADMIN_USERS::FORM_DESCRIPTION_INPUT_VALUE';
+export const FORM_USERNAME_INPUT_VALUE = 'ADMIN_USERS::FORM_USERNAME_INPUT_VALUE';
+export const FORM_PASSWORD_INPUT_VALUE = 'ADMIN_USERS::FORM_PASSWORD_INPUT_VALUE';
+export const FORM_PASSWORD_REPEAT_INPUT_VALUE = 'ADMIN_USERS::FORM_PASSWORD_REPEAT_INPUT_VALUE';
+export const PASSWORD_LENGTH_ERROR = 'ADMIN_USERS::PASSWORD_LENGTH_ERROR';
+export const PASSWORD_EQUAL_ERROR = 'ADMIN_USERS::PASSWORD_EQUAL_ERROR';
+export const EMPTY_FIELDS_ERROR = 'ADMIN_USERS::EMPTY_FIELDS_ERROR';
+export const CLOSE_ACTION = 'ADMIN_USERS::CLOSE_ACTION';
```
Comment (VladaZhmurkevych) [inline src/store/adminUsers/types.js:22]: I think it would be better to create another store named 'createUser' and write everything there

File: src/store/connection/types.js [hunk 19]
```diff
@@ -9,5 +9,14 @@ export const CREATE_SUCCESS = 'ADMIN_CONNECTIONS::CREATE_SUCCESS';
...
 export const CHANGE_SEARCH_INPUT = 'ADMIN_CONNECTIONS::CHANGE_SEARCH_INPUT';
+
+export const HOST_VALUE = 'ADMIN_CONNECTIONS::HOST_VALUE';
+export const PORT_VALUE = 'ADMIN_CONNECTIONS::PORT_VALUE';
+export const NAME_DB_VALUE = 'ADMIN_CONNECTIONS::NAME_DB_VALUE';
+export const NAME_CONNECTION_VALUE = 'ADMIN_CONNECTIONS::NAME_СONNECTION_VALUE';
+export const USERNAME_VALUE = 'ADMIN_CONNECTIONS::USERNAME_VALUE';
+export const PASSWORD_VALUE = 'ADMIN_CONNECTIONS::PASSWORD_VALUE';
+export const TYPE_VALUE = 'ADMIN_CONNECTIONS::TYPE_VALUE';
+export const CLOSE_ACTION = 'ADMIN_USERS::CLOSE_ACTION';
```
Comment (VladaZhmurkevych) [inline src/store/connection/types.js:20]: the same as for users, it is better to create a new store and write it there, as it is not very related to connections, it more related to form

File: src/store/connection/actions.js (hunk 10 above)
Comment (delieva) [reply src/store/connection/actions.js]: There is no effect on the popup, apart from demonstrating error text at the bottom

**Review by VladaZhmurkevych — CHANGES_REQUESTED** [review]

File: src/components/shared/BasePopup/base.popup.module.scss (hunk 16 above)
Comment (delieva) [reply src/components/shared/BasePopup/base.popup.module.scss:39]: it needs to, don't you think that this numbers are better ones?

File: src/components/newDBConnectionPopup/index.js [hunk 20]
```diff
@@ -0,0 +1,189 @@
...
+  return (
+    <CSSTransition
+      in={isVisible}
+      timeout={300}
+      classNames="alert"
+      unmountOnExit
+    >
+      <div className={styles.popupWrapper}>
+        <div className={styles.blur} onClick={backgroundHandler} />
+        <div className={`${styles.popup}`}>
```
Comment (VladaZhmurkevych) [inline src/components/newDBConnectionPopup/index.js:81]: What is it?

File: src/store/connection/types.js (hunk 19 above)
Comment (delieva) [reply src/store/connection/types.js:20]: Are you talking about types.js or about all four files in store/connection?

File: src/store/adminUsers/actions.js [hunk 21]
```diff
@@ -34,6 +55,14 @@ export const createUserSuccess = createAction(CREATE_SUCCESS, (user) => user);
 export const setUsers = createAction(SET_USERS, (users) => users);
 export const appendUsers = createAction(APPEND_USERS, (users) => users);
 export const changeSearchInput = createAction(CHANGE_SEARCH_INPUT, (value) => value);
+export const passwordLengthError = createAction(PASSWORD_LENGTH_ERROR, (value) => value);
+export const passwordEqualError = createAction(PASSWORD_EQUAL_ERROR, (value) => value);
+export const getUserTypeValue = createAction(FORM_USER_TYPE_INPUT_VALUE, (object) => object.target.value);
+export const getUsernameValue = createAction(FORM_USERNAME_INPUT_VALUE, (value) => value);
+export const getPasswordValue = createAction(FORM_PASSWORD_INPUT_VALUE, (value) => value);
+export const getUserDescriptionValue = createAction(FORM_DESCRIPTION_INPUT_VALUE, (value) => value);
+export const getPasswordRepeatValue = createAction(FORM_PASSWORD_REPEAT_INPUT_VALUE, (value) => value);
```
Comment (MishaHnatyshyn) [inline src/store/adminUsers/actions.js:66]: Why do these actions start with `get` ?

File: src/components/newDBConnectionPopup/index.js (hunk 20 above)
Comment (delieva) [reply src/components/newDBConnectionPopup/index.js:81]: that is the code from base popup that I duplicated. I'm rewriting code to use basepopup itself

File: src/components/shared/BasePopup/base.popup.module.scss (hunk 17 above)
Comment (VladaZhmurkevych) [reply src/components/shared/BasePopup/base.popup.module.scss]: It is better to create a file with exactly your styles to the specific popup and write them there

File: src/store/connection/types.js (hunk 19 above)
Comment (VladaZhmurkevych) [reply src/store/connection/types.js:20]: I am talking about all files

File: src/store/adminUsers/actions.js (hunk 21 above)
Comment (delieva) [reply src/store/adminUsers/actions.js:66]: do you prefer another start? Like `change`?

File: src/components/shared/BasePopup/base.popup.module.scss (hunk 16 above)
Comment (VladaZhmurkevych) [reply src/components/shared/BasePopup/base.popup.module.scss:39]: BasePopup is used for errors popup too, and there is no need to create big pop up, maybe it would be better to create some wrapper for forms based on basePopUp and set this style there? Or to pass it into component, like "classes".

File: src/store/connection/types.js (hunk 19 above)
Comment (delieva) [reply src/store/connection/types.js:20]: We discussed it with Misha Hnatyshyn. He sad this logic should be implemented here

File: src/components/newDBConnectionPopup/index.js (hunk 20 above)
Comment (VladaZhmurkevych) [reply src/components/newDBConnectionPopup/index.js:81]: Cool

File: src/store/adminUsers/actions.js (hunk 21 above)
Comment (VladaZhmurkevych) [reply src/store/adminUsers/actions.js:66]: Yes, change is better
